File: 2358-number-of-ways-to-split-array/number-of-ways-to-split-array.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:

        s=sum(nums)
        valid=0
        n=len(nums)
        s1=0
        for i in range(n-1):
            s1+=nums[i]
            if s1>= (s-s1):
                valid+=1
            

        return valid 


        

File: 2358-number-of-ways-to-split-array/number-of-ways-to-split-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var waysToSplitArray = function(nums) {


    function sum()
    {
        let s=0;
        for(let i=0;i<nums.length;i++)
        {
            s+=nums[i];

        }
        return s;
    }

    const s1=sum();
    let s2=0;
    let valid=0;
    for(let i=0;i<nums.length-1;i++)
    {

        s2+=nums[i];

        if(s2>=(s1-s2))
        {
            valid++;

        }

    }

    return valid;
    
};